### PR TITLE
lightning: move tests out of br

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #52182 

Problem Summary:

### What changed and how does it work?
lightning ci must be run with this https://github.com/PingCAP-QE/ci/pull/2882, so need merge them together
- br ci pass https://github.com/pingcap/tidb/pull/52289#issuecomment-2030024426
- lightning ci pass  https://github.com/pingcap/tidb/pull/52289#issuecomment-2030958603
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by D3Hunter** [comment]
br ci pass https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_br_integration_test/detail/pull_br_integration_test/97/pipeline

File: br/tests/lightning_checksum_mismatch/run.sh
Comment (D3Hunter) [inline br/tests/lightning_checksum_mismatch/run.sh:1]: they are renamed actually, but with license added, they are taken as `delete/new`

**Comment by D3Hunter** [comment]
lightning ci must be run with this https://github.com/PingCAP-QE/ci/pull/2882, so it's replayed in here https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_lightning_integration_test/detail/pull_lightning_integration_test/149/pipeline

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
I found `br/pkg/lightning/checkpoints` is not deleted in master branch, can you delete it in this PR?

**Comment by D3Hunter** [comment]
> The full list of commands accepted by this bot can be found [here](https://prow.tidb.net/command-help?repo=pingcap%2Ftidb).

local empty dir? it's moved away already https://github.com/pingcap/tidb/tree/master/br/pkg

**Review by Leavrth — APPROVED** [review]

**Pull request merged**